src/components/auth/PrivateRoute.tsx

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import {useSelector} from 'react-redux';
+import {Route, Redirect, RouteProps} from 'react-router-dom';
+
+import {RootState} from '../../store';
+
+interface Props extends RouteProps {
+    component: any;
```

Comment:
No `any` everywhere.

And because you want it to be a component, type annotates it like `component: Component`.